proxy/pkg/config/config.go

```diff
@@ -133,12 +135,46 @@ func New() *Config {
 	return &Config{}
 }
 
+func (c *Config) loadFromFiles() error {
+	paths := os.Getenv("ZDM_CONFIG_FILES")
+	for _, path := range strings.Split(paths, ",") {
+		if len(path) == 0 {
+			continue
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("could not read configuration file %v: %w", path, err)
+		}
+		defer file.Close()
+
+		dec := yaml.NewDecoder(file)
+		if err = dec.Decode(c); err != nil {
```

Comment:
what happens if the same config settings are defined in multiple files? As it stands it seems that they will be overwritten which is not a very clear behavior. We should probably return an error and fail the launch if that happens.

Reply:
It was intentional to allow to override. At the end we already print all applied configuration variables, so there should be no confusion.

Reply:
I heavily disagree with this, I don't see any benefit to it and it might just lead to user error, users never check the configuration that is printed in the logs, it's mostly for us devs to troubleshoot issues.

Reply:
I hadn't thought about the possibility of the same value being set in multiple files, but it is a good point and definitely a risk.

I would see the value in supporting multiple files so that users can have one with stricter governance and another with easier access. For example, they may want to have one file containing credentials / SCB path / contact points, with restricted access, and another with things like read mode, primary cluster, log level etc etc which can be more easily accessed.

This could also be managed by having clearly structured and named files, from which we extract only the expected properties. This would not leave it up to the user to define any property in any file, and would ensure that each property is defined and considered exactly once. The downside is that we need to parse each file specifically for the expected content, but the configuration properties themselves do not tend to change, so I wouldn't see it as a big problem.

Reply:
I wonder if we are overengeneering this a bit... In general most tools just allow you to provide a config file and that's it. In the case of DSE, DSE allows you to set a config setting that is a path to a key file so that DSE can decrypt the credentials.

Can we take a step back and actually think if we need multiple config files? I feel like the DSE approach is a simple one but effective no?